gtsfm/loader/loader_base.py

```diff
@@ -81,6 +85,77 @@ def is_valid_pair(self, idx1: int, idx2: int) -> bool:
             validation result.
         """
 
+    def get_image(self, index: int) -> Image:
+        """Get the image at the given index, satisfying a maximum image resolution constraint.
+
+        Determine how the camera intrinsics and images should be jointly rescaled based on desired img. resolution.
+        Each loader implementation should set a `_max_resolution` attribute.
+
+        Args:
+            index: the index to fetch.
+
+        Raises:
+            IndexError: if an out-of-bounds image index is requested.
+
+        Returns:
+            Image: the image at the query index. It will be resized to satisfy the maximum
+                allowed loader image resolution if the full-resolution images for a dataset
+                are too large.
+        """
+        if not hasattr(self, "_max_resolution"):
```

Comment:
We should ideally fail during object construction rather than here. Maybe by calling the constructor for the parent class?

Reply:
should be fixed now